src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPFoldingRangeFeature.java

```diff
@@ -62,4 +65,42 @@ public void setServerCapabilities(@Nullable ServerCapabilities serverCapabilitie
             foldingRangeCapabilityRegistry.setServerCapabilities(serverCapabilities);
         }
     }
+
+    /**
+     * Whether or not the specified folding range should be collapsed by default. The default implementation returns
+     * true for folding ranges when:
+     * <ul>
+     *     <li><code>kind</code> is {@link FoldingRangeKind#Imports} and {@link CodeFoldingSettings#COLLAPSE_IMPORTS}
+     *         is true
+     *     </li>
+     *     <li><code>kind</code> is {@link FoldingRangeKind#Comment}, {@link CodeFoldingSettings#COLLAPSE_FILE_HEADER}
+     *         is true, and the folding range starts on the file's first line
+     *     </li>
+     * </ul>
+     *
+     * @param file         the PSI file
+     * @param foldingRange the folding range
+     * @return true if the folding range should be collapsed by default; otherwise false
+     */
+    public boolean isCollapsedByDefault(@NotNull PsiFile file, @NotNull FoldingRange foldingRange) {

```

Comment:
The default implementation is exactly as it was in the other location and is documented well here.